store/store.go

```diff
@@ -662,6 +663,68 @@ func (c *Client) GetDataV1(ctx context.Context, traceID string, userID string, d
 	return dataCollection(c).Find(ctx, query, opts)
 }
 
+/* Get the information about IMEI device, it takes a subset of the detailed information from pumpsettings
+db.deviceData.aggregate([
+	{$match: { "payload.device.imei": "1234567890", type: "pumpSettings"}},
+	{$project:{
+				_userId: 1,
+				device: "$payload.device",
+				pump: "$payload.pump",
+				cgm: "$payload.cgm",
+				time: 1,
+				uploadId: 1
+				}
+	},
+	{$sort:  { time : -1}},
+	{$limit: 5},
+])
+*/
+func (c *Client) GetImeiV1(ctx context.Context, traceID string, imei string, limit int, dates *Date) (goComMgo.StorageIterator, error) {
+
+	match := bson.M{
+		"payload.device.imei": imei,
+		"type":                "pumpSettings",
+	}
+
+	if dates.Start != "" && dates.End != "" {
+		match["time"] = bson.M{"$gte": dates.Start, "$lt": dates.End}
+	} else if dates.Start != "" {
+		match["time"] = bson.M{"$gte": dates.Start}
+	} else if dates.End != "" {
+		match["time"] = bson.M{"$lt": dates.End}
+	}
+
+	pipeline := []bson.M{
```

Comment:
What is the advantage of doing an aggregate pipeline here instead of just a find().sort().limit in that case ? 
Just renaming payload.device/pump/cgm ?